lab_1/main.py

```diff
@@ -3,16 +3,23 @@
 A concordance extraction
 """
 
+import re
+def tokenize():
+    word = ''
+    f = open('data.txt', 'r')
```

Comment:
You do not need to open it here: there is a function `read_from_file` which can do just that.